### PR TITLE
Tdl 14963 removed endpoints dict

### PR DESCRIPTION
**Pull request by namrata270998**
# Description of change

- removed the endpoints dictionary
- added an Endpoints Class

# Manual QA steps
 - Check if the endpoint API URL is correct for each stream.
 - Check that the new code generates the same catalog files as before
 - Check that the new code generates the same records with and without the state file as before.

 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by dbshah1212 — APPROVED** [review]

**Comment by dbshah1212** [comment]
@namrata270998 Make sure you merge this code after back merging the tap-tester code and all tests passed.

File: tap_pendo/streams.py
Comment (karanpanchal-crest) [inline tap_pendo/streams.py:79]: @namrata270998 Add required unit test cases for these code changes

Comment (namrata270998) [reply tap_pendo/streams.py:79]: Done

File: tap_pendo/streams.py
Comment (karanpanchal-crest) [inline tap_pendo/streams.py:92]: @namrata270998 Add comments to this function

File: tap_pendo/streams.py
Comment (karanpanchal-crest) [inline tap_pendo/streams.py:102]: @namrata270998 Add comment to this code

File: tap_pendo/streams.py
Comment (karanpanchal-crest) [inline tap_pendo/streams.py:534]: @namrata270998 Why did we remove this attribute? Isn't it used anywhere?

File: tap_pendo/streams.py
Comment (karanpanchal-crest) [inline tap_pendo/streams.py:856]: @namrata270998 Is this function being used anywhere?

File: tap_pendo/streams.py
Comment (karanpanchal-crest) [inline tap_pendo/streams.py:925]: @namrata270998 Please add unit test case for this code change

File: tap_pendo/streams.py
Comment (karanpanchal-crest) [inline tap_pendo/streams.py:898]: Add unit test case for this code change

File: tap_pendo/streams.py
Comment (karanpanchal-crest) [inline tap_pendo/streams.py:816]: Add comments to this code change.
Also, can you please explain, how would the 'VisitorId' parameter be assigned a value?

File: tap_pendo/streams.py
Comment (karanpanchal-crest) [inline tap_pendo/streams.py:784]: Add unit test case for this code change

File: tests/unittests/test_endpoints_dict.py
Comment (karanpanchal-crest) [inline tests/unittests/test_endpoints_dict.py:1]: @namrata270998 Did we execute the integration test cases for the same?

**Review by karanpanchal-crest — APPROVED** [review]

File: tap_pendo/streams.py
Comment (namrata270998) [reply tap_pendo/streams.py:784]: Have added one common unit test case for this change in the test_endpoints_dict.py file

File: tap_pendo/streams.py
Comment (namrata270998) [reply tap_pendo/streams.py:908]: No, removed it

File: tap_pendo/streams.py
Comment (namrata270998) [reply tap_pendo/streams.py]: No, removed it

File: tap_pendo/streams.py
Comment (namrata270998) [reply tap_pendo/streams.py:828]: Done.
The get_url function of the Endpoints class will be used, and in that function this `visitorId` will be formatted and replaced with the original one

**Pull request closed without merging**
